cas-server-core-api-authentication/src/main/java/org/jasig/cas/authentication/RootCasException.java

```diff
@@ -12,7 +12,7 @@
  * see org.jasig.cas.authentication.handler.AuthenticationException
  * @since 4.0.0
  */
-public abstract class RootCasException extends Exception {
+public abstract class RootCasException extends RuntimeException {
```

Comment:
I agree that it does cause changes in signatures which are undesirable. But at the same time those signature changes force code using those interfaces to account for new exceptions being thrown from certain execution paths before an unexpected Exception is thrown in production due to an untested corner case exercising that exception.

Internally i feel it also helps us account for exceptions we may not be thinking about when we create new methods that need to create a PGT before they lead to unexpected exceptions at runtime.

I don't feel that all exceptions should be checked, exceptions that arise from improper usage should be unchecked. Or exceptions not used to convey expected failure scenarios.

This change cascades consequences to usage of the `CentralAuthenticationService` object which throws `AbstractTicketException` from (all?) methods. Now if we are implementing a new component that needs to grant a `ST` from a `TGT` we will need to remember to account for the `InvalidTicketException` since it's no longer a checked exception that is being forced on us.

Maybe an alternative is to have the new `InvalidProxyGrantingTicketForServiceTicket` exception not extend the current `AbstractTicketException` hierarchy, but extend a separate `RuntimeException` hierarchy.

Anyways, I feel I've made my case, so I'll leave the decision up to you :)